### PR TITLE
Add some parallelism to file cache linking.

### PR DESCRIPTION
**Pull request by vadimberezniker**
In my tests, it increases performance by more than 5x.

**Related issues**: N/A

### CODE REVIEW

File: enterprise/server/remote_execution/dirtools/dirtools.go [hunk 1]
```diff
@@ -680,16 +718,27 @@ func (ff *BatchFileFetcher) FetchFiles(filesToFetch FileMap, opts *DownloadTreeO
...
 
-	currentBatchRequestSize := int64(0)
-	eg, ctx := errgroup.WithContext(ff.ctx)
+	linkEG, _ := errgroup.WithContext(ff.ctx)
+	if runtime.GOOS == "darwin" {
+		// Macs don't appear to like too much parallelism, with overall
+		// performance dropping with high parallelism.
+		linkEG.SetLimit(5)
+	} else {
+		linkEG.SetLimit(100)
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/dirtools/dirtools.go:728]: does it make sense to max out at runtime.GOMAXPROCS(0)?

File: enterprise/server/remote_execution/dirtools/dirtools.go [hunk 2]
```diff
@@ -702,67 +751,76 @@ func (ff *BatchFileFetcher) FetchFiles(filesToFetch FileMap, opts *DownloadTreeO
...
+			fetchQueue <- digestToFetch{d: d, fps: filePointers}
+			return nil
+		})
+	}
+
+	// Read work off the fetchQueue channel and generate batch read requests
+	// to download data.
+	eg, ctx := errgroup.WithContext(ff.ctx)
+	enqueueWorkDone := make(chan struct{})
+	go func() {
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/dirtools/dirtools.go:771]: did you consider using the errgroup for this goroutine as well? That might let you get rid of enqueueWorkDone and just wait on the eg

**Review by tylerwilliams — APPROVED** [review]

**Comment by tylerwilliams** [comment]
Also: great find :) This is really cool

File: enterprise/server/remote_execution/dirtools/dirtools.go (hunk 1 above)
Comment (vadimberezniker) [reply enterprise/server/remote_execution/dirtools/dirtools.go]: done

**Pull request merged**